components/interface/TreeWidget.js

```diff
@@ -420,34 +431,76 @@ export default class TreeWidget extends React.Component {
 
   getButtons (rowInfo) {
     var buttons = [];
-    if (rowInfo.node.title !== "No data available.") {
-      buttons.push(<i className="fa fa-tint"
-        aria-hidden="true"
-        onClick={ () => {
-          this.setState({ displayColorPicker: true });
-        }}>
-        { (this.state.displayColorPicker
+    if (Instances[rowInfo.node.instanceId] !== undefined && typeof Instances[rowInfo.node.instanceId].isVisible !== "undefined") {
+      // rowInfo.node.subtitle = rowInfo.node.instanceId;
+      if (Instances[rowInfo.node.instanceId].isVisible()) {
+        var color = Instances[rowInfo.node.instanceId].getColor();
+        buttons.push(<i className="fa fa-eye-slash"
+          aria-hidden="true"
+          onClick={ () => {
+            Instances[rowInfo.node.instanceId].hide();
+            this.setState({ nodeSelected: rowInfo.node });
+          }} />);
+        buttons.push(<i className="fa fa-tint"
+          style={{
+            paddingLeft: "6px",
+            color: color
+          }}
+          aria-hidden="true"
+          onClick={ () => {
+            this.setState({ displayColorPicker: true });
+          }}>
+          { (this.state.displayColorPicker
```

Comment:
@ddelpiano ternary operator here makes the code hard to read, it's ok for short one-liners but in cases like this I would avoid it